proto/message.py

```diff
@@ -369,16 +373,95 @@ def deserialize(cls, payload: bytes) -> "Message":
+            PROTOBUF_VERSION[0] not in ("3", "4")
+            and including_default_value_fields is not None
+        ):
+            warnings.warn(
+                """The argument `including_default_value_fields` has been removed from
+                Protobuf 5.x. Please use `always_print_fields_with_no_presence` instead.
+                """,
+                DeprecationWarning,
+            )
+
+    def _raise_if_print_fields_values_are_set_and_differ(
+        cls,
+        always_print_fields_with_no_presence: Optional[bool],
+        including_default_value_fields: Optional[bool],
+    ) -> None:
+        """
+        Raise Exception if both `always_print_fields_with_no_presence` and `including_default_value_fields` are set
+            and the values differ.
+
+        Args:
+            always_print_fields_with_no_presence (Optional(bool)): The value of `always_print_fields_with_no_presence` set by the user.
+            including_default_value_fields (Optional(bool)): The value of `including_default_value_fields` set by the user.
+        Returns:
+            None
+        Raises:
+            ValueError: if both `always_print_fields_with_no_presence` and `including_default_value_fields` are set and
+                the values differ.
+        """
+        if (
+            always_print_fields_with_no_presence is not None
+            and including_default_value_fields is not None
+            and always_print_fields_with_no_presence != including_default_value_fields
+        ):
+            raise ValueError(
+                "Arguments `always_print_fields_with_no_presence` and `including_default_value_fields` must match"
+            )
+
+    def _normalize_print_fields_without_presense(
+        cls,
+        always_print_fields_with_no_presence: Optional[bool],
+        including_default_value_fields: Optional[bool],
+    ) -> bool:
+        """
+        Return true if fields with no presense should be included in the results.
+        By default, fields with no presense will be included in the results
+        when both `always_print_fields_with_no_presence` and
+        `including_default_value_fields` are not set
+
+        Args:
+            always_print_fields_with_no_presence (Optional(bool)): The value of `always_print_fields_with_no_presence` set by the user.
+            including_default_value_fields (Optional(bool)): The value of `including_default_value_fields` set by the user.
+        Returns:
+            None
+        Raises:
+            ValueError: if both `always_print_fields_with_no_presence` and `including_default_value_fields` are set and
+                the values differ.
+        """
+        if (
+            always_print_fields_with_no_presence is None
+            and including_default_value_fields is None
+        ):
+            print_fields = True
+        else:
+            print_fields = (
+                always_print_fields_with_no_presence or including_default_value_fields
+            )
```

Comment:
The suggestion didn't work because we need to use a default of `print_fields=True` if both `always_print_fields_with_no_presence` and `including_default_value_fields` are `None` (not set).  

We need to distinguish between not set and `False`, especially since we are raising an exception if `always_print_fields_with_no_presence` and `including_default_value_fieldsincluding_default_value_fields` are set and don't match.

Reply:
Ah, gotcha. I missed that part.  I think the following would do it, though it's arguably less clear than what you have.

```suggestion
        print_fields = (
            always_print_fields_with_no_presence is None
            and including_default_value_fields is None
        ) or (always_print_fields_with_no_presence or 
              including_default_value_fields)
```

Reply:
(It's always weird to have 
```python
if x:
  y = true
```

We could also use a pattern a bit more explicit than my suggestion above
```python
y=x
x=x or (...)
```

Reply:
Fixed in https://github.com/googleapis/proto-plus-python/pull/433/commits/4e9f90e14104987984f0825fbba4a43dae8dff9b